pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -369,11 +369,49 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 
 				// Increase BuildRun count in metrics
 				buildmetrics.BuildRunCountInc(buildRun.Status.BuildSpec.StrategyRef.Name)
-				// Add BuildRun establish and completion times in metrics
-				buildRunCompletionDuring := buildRun.Status.CompletionTime.Time.Sub(buildRun.CreationTimestamp.Time)
-				buildRunEstablishDuring := buildRun.Status.StartTime.Time.Sub(buildRun.CreationTimestamp.Time)
-				buildmetrics.BuildRunEstablishObserve(buildRun.Status.BuildSpec.StrategyRef.Name, buildRun.Namespace, buildRunEstablishDuring)
-				buildmetrics.BuildRunCompletionObserve(buildRun.Status.BuildSpec.StrategyRef.Name, buildRun.Namespace, buildRunCompletionDuring)
+
+				// Time between the creation of the buildrun and the start of the buildrun (buildrun ramp-up)
+				buildmetrics.BuildRunEstablishObserve(
+					buildRun.Status.BuildSpec.StrategyRef.Name,
+					buildRun.Namespace,
+					buildRun.Status.StartTime.Time.Sub(buildRun.CreationTimestamp.Time),
+				)
+
+				// Total time between the creation of the buildrun and the buildrun completion
+				buildmetrics.BuildRunCompletionObserve(
+					buildRun.Status.BuildSpec.StrategyRef.Name,
+					buildRun.Namespace,
+					buildRun.Status.CompletionTime.Time.Sub(buildRun.CreationTimestamp.Time),
+				)
+
+				// Look for the pod created by the taskrun
+				var pods = &corev1.PodList{}
+				if err := r.client.List(ctx, pods, client.InNamespace(request.Namespace), client.MatchingLabels{"buildrun.build.dev/name:": buildRun.Name}); err == nil && len(pods.Items) == 1 {
```

Comment:
This list call imo can be replaced by a client.Get. You get the pod name from `taskRun.Status.PodName`

Reply:
Good point, I changed the implementation.